pkg/controller/util/endpoint/deduplicating_rate_limiting_queue.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+// DeduplicatingRateLimitingQueue implements a wrapper on workqueue.RateLimitingInterface that also rate limits initial key additions
```

Comment:
Why does the name have the word "deduplicating" in it, given the description is about using it for everything and not just failures?

Reply:
Yeah, the name should be changed probably. The best I can figure out is simply RateLimitingQueue but it's used already by workqueue package with a different semantics. Any suggestion?